lightcurve/static/lightcurve/js/lightcurve.js

```diff
@@ -74,18 +73,20 @@ $(function() {
     return false;
   });
 
-  $(document).on('click', '.edit-name', function(e) {
-    e.preventDefault();
+  $('.js-edit-name').on('click', function() {
     var filename = $(this).prev(".lightcurve-name").text();
     $( $(this).prev(".lightcurve-name") ).replaceWith( "<input type='name' class='input_name' name='input_name' value='" + filename + "'>" );
-    $(this).replaceWith("<button type='submit' class='btn btn-sm btn-primary submit-name'>Save</button><button class='btn btn-sm btn-default cancel'>Cancel</button>");
+    $(this).css('display', 'none');
+    $(this).nextAll('button').css('display', 'inline-block');
```

Comment:
I don't think we should use `next` or `parent` selectors, in general.  This is brittle because purely aesthetic UI changes can break JS code that depends on positiong.  It's best to address them directly by eg. classname.

Reply:
Done.